pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -846,6 +902,20 @@ func (pl *DynamicResources) bindClaim(ctx context.Context, state *stateData, ind
 				claim = updatedClaim
 			}
 			claim.Status.Allocation = allocation
+
+			// if the claim has devices with binding conditions, we need to store them in the claim status
+			if pl.enableDeviceBindingConditions {
```

Comment:
I don't think we need this section. We should not be writing the status. Just wait for the driver to create it.